shared/teams/team/invites/generate-link.tsx

```diff
@@ -0,0 +1,364 @@
+      setRolePickerOpen(false)
+    },
+    onOpenRolePicker: () => setRolePickerOpen(true),
+    onSelectRole: (role: Types.TeamRoleType) => setTeamRole(role),
+    teamRole: teamRole,
+  }
+
+  if (inviteLink != null) {
+    return (
+      <Kb.Modal
+        onClose={onClose}
+        header={{
+          leftButton: <Kb.Icon type="iconfont-arrow-left" onClick={onBack} />,
+          title: <ModalTitle teamID={teamID} title="Share an invite link" />,
+        }}
+        footer={{
+          content: (
+            <Kb.Button
+              fullWidth={true}
+              label="Close"
+              onClick={inviteLink.expired ? onClose : onGenerate}
+              type="Dim"
+            />
+          ),
+          hideBorder: true,
+        }}
+        allowOverflow={true}
+      >
+        <Kb.Box2 direction="horizontal" fullWidth={true} style={styles.banner} centerChildren={true}>
+          {inviteLink.expired ? (
+            <Kb.Icon type="icon-illustration-teams-invite-links-grey-460-96" />
+          ) : (
+            <Kb.Icon type="icon-illustration-teams-invite-links-green-460-96" />
+          )}
+        </Kb.Box2>
+        <Kb.Box2
+          direction="vertical"
+          fullWidth={true}
+          style={styles.body}
+          gap={Styles.isMobile ? 'xsmall' : 'tiny'}
+        >
+          <Kb.Text type="Body" style={styles.infoText}>
+            Here is your link. Share it cautiously as anyone who has it can join the team.
+          </Kb.Text>
+
+          <InviteItem
+            duration={inviteDuration}
+            inviteLink={inviteLink}
+            onExpireCallback={onExpire}
+            teamID={teamID}
+          />
+
+          {inviteLink.expired && <Kb.Text type="BodySmallSemiboldPrimaryLink">Generate a new link</Kb.Text>}
+        </Kb.Box2>
+      </Kb.Modal>
+    )
+  }
+
+  return (
+    <Kb.Modal
+      onClose={onClose}
+      header={{
+        leftButton: <Kb.Icon type="iconfont-arrow-left" onClick={onBack} />,
+        title: <ModalTitle teamID={teamID} title="Share an invite link" />,
+      }}
+      footer={{
+        content: <Kb.Button fullWidth={true} label="Generate invite link" onClick={onGenerate} />,
+        hideBorder: true,
+      }}
+      allowOverflow={true}
+    >
+      <Kb.Box2 direction="horizontal" fullWidth={true} style={styles.banner} centerChildren={true}>
+        <Kb.Icon type="icon-illustration-teams-invite-links-blue-460-96" />
+      </Kb.Box2>
+      <Kb.Box2
+        direction="vertical"
+        fullWidth={true}
+        style={styles.body}
+        gap={Styles.isMobile ? 'xsmall' : 'tiny'}
+      >
+        <Kb.Text type="Body" style={styles.infoText}>
+          Invite people to {teamname} by sharing a link:
+        </Kb.Text>
+
+        <Kb.Box2 direction={Styles.isMobile ? 'vertical' : 'horizontal'} fullWidth={true}>
+          <Kb.Text type="BodySmall" style={styles.rowTitle}>
+            Validity
+          </Kb.Text>
+          <Kb.Dropdown
```

Comment:
In designs this is attached to a `FloatingMenu`. changing it to an inlinedropdown + that should fix the alignment problem and make creating these items less clunky.